### PR TITLE
[#8] Implement Typhoeus, move to libcurl and add keepalive

### PR DESCRIPTION
**Pull request by mattrayner**
The aim of this pull request is a couple of things:

* Implement #8, as part of a performance improvement at the client level
* Move to Typhoeus as an underlying library. rest_client is great, but moving away from Net::HTTP to a libcurl foundation will provide a sizeable boost on it's own
* Includes the latest macmillan-utils stuff

Let me know what you think?

### CODE REVIEW

File: spec/lib/bandiera/client_spec.rb [hunk 1]
```diff
@@ -57,12 +57,12 @@
...
-      response = double(:response, body: JSON.generate({ response: {} }))
-      resource = double(:resource, '[]' => double(get: response))
+      response = double(:response, body: JSON.generate({ response: {} }), success?: true)
+      resource = double(:resource, run: response)
 
-      expect(RestClient::Resource)
+      expect(Typhoeus::Request)
         .to receive(:new)
-        .with(api_uri, method: :get, timeout: 24, open_timeout: 24, headers: anything)
+        .with("#{api_uri}/v2/groups/foo/features/bar", method: :get, timeout: 24, open_timeout: 24, headers: {"User-Agent"=>"Bandiera Ruby Client / 3.0.4"}, params: {})
```
[the remark this answers is not in the document]
Comment (dazoakley) [reply spec/lib/bandiera/client_spec.rb]: `Place the . on the previous line` - bugger off hound :D

File: .travis.yml [hunk 2]
```diff
@@ -7,7 +7,9 @@ rvm:
   - 2.0
   - 2.1
   - 2.2
-  - 2.3.0
+  - 2.3
+  - 2.4
+  - 2.5
```
Comment (dazoakley) [inline .travis.yml]: awesome, thanks for this!

File: spec/lib/bandiera/client_spec.rb [hunk 3]
```diff
@@ -53,23 +53,64 @@
...
```
Comment (dazoakley) [inline spec/lib/bandiera/client_spec.rb]: Is there a way to auto-generate the user agent string?  As I can see this test failing when we bump the version...

**Comment by mattrayner** [comment]
@dazoakley I've made the change to the client test as per your request and tested it locally with a version change 👍

Thanks for reviewing so quickly

**Comment by mattrayner** [comment]
Also, that 'place . on the line above' has got to be a joke 😂

**Comment by mattrayner** [comment]
@dazoakley how do we go about getting this merged in/a new version released - is there another person that needs to review?

**Comment by dazoakley** [comment]
LOL! D'oh! That's totally me - I could/should have done that earlier. :D

Not near the computer tonight, but will sort first thing tomorrow for ya. :+1:

**Pull request merged**